flow2ml/Tf_Results.py

```diff
@@ -0,0 +1,129 @@
+
+class Tf_Results:
+    def __init__(self,model,validtion_generator):
+        '''
+        Initialises the class variables.
+        '''
+        self.model = model          #stores the model
+        self.validation_generator = validtion_generator    
+        try:                        #try creating a Results directory and show excpetion if it already exists 
+            p = os.getcwd()
+            p = os.path.join(p,"Results")
+            self.results_path = p
+            os.mkdir(self.results_path)
+        except:
+            print("Results Folder already exists")
+
+    def tf_roc_curve(self,figure_name="roc.jpeg",labels=None):
+        '''
+        Plotting the Roc curve for the Input model
+        '''
+        if labels == None:
+            labels = self.validation_generator.classes
+        preds = self.model.predict(self.validation_generator,verbose=1)
+        fpr, tpr, _ = roc_curve(labels, preds)
+        roc_auc = auc(fpr, tpr)
+        lw = 2
+        plt.figure(figsize=(50, 20))
+        plt.plot(fpr, tpr, color='darkorange',
+        lw=lw, label='ROC curve (area = %0.2f)' % roc_auc)
+        plt.plot([0, 1], [0, 1], color='navy', lw=lw, linestyle='--')
+        plt.xlabel('False Positive Rate')
+        plt.ylabel('True Positive Rate')
+        plt.title('Receiver operating characteristic')
+        plt.legend(loc="lower right")
+        #plot_roc_curve(self.model, self.x_test, self.y_test) 
+        #plt.title('ROC Curve')
+        plt.savefig(self.results_path+'/'+figure_name)
+    
+    def tf_confusion_matrix(self):
+        '''
+        Plotting the confusion matrix
+        '''
+        preds = self.model.predict(self.validation_generator,verbose=1)
+        labels = self.validation_generator.classes
+        df_cm = tf.math.confusion_matrix(
+    labels, preds, num_classes=None, weights=None, dtype=tf.dtypes.int32,
+    name=None)
+        plt.figure(figsize=(50,20))
+        sn.heatmap(df_cm, annot=True, annot_kws={"size": 35})
+        plt.title('Confusion Matrix', size = 35)
+        plt.savefig(self.results_path+'/confusion_matrix.jpeg')
+    
+    def tf_precision_recall_curve(self,figure_name="prc.jpeg",labels=None):
+        '''
+        Plotting the precision recall curve 
+        '''
+        if labels == None:
+            labels = self.validation_generator.classes
+        preds = self.model.predict(self.validation_generator,verbose=1)
+        pr, tpr, _ = precision_recall_curve(labels, preds)
+        lw = 2
+        plt.figure(figsize=(50,20))
+        plt.plot(tpr, pr, color='darkorange',
+        lw=lw)
+        plt.xlabel('Recall')
+        plt.ylabel('Precision')
+        plt.title('Precision Recall Curve')
+        
+        # plot_precision_recall_curve(self.model,self.x_test,self.y_test)
+        # plt.title('Precision Recall Curve', size = 35)
+        plt.savefig(self.results_path+'/'+figure_name)
+
+    def tf_get_results_docx(self,file_name="report.docx"):
+        '''
```

Comment:
What if already there is a file with name “report.docx”

Reply:
the filename can be changed by the user.

Reply:
Yes, okay